client/goal.c

```diff
@@ -667,3 +670,93 @@ TDNFSolvAddPkgLocks(
     goto cleanup;
 }
 
+uint32_t
+TDNFSolvAddMinVersions(
+    PTDNF pTdnf,
+    Queue* pQueueJobs,
+    Pool *pPool
+    )
+{
+    uint32_t dwError = 0;
+    char **ppszPackages = NULL;
+    char *pszPkg = NULL;
+    char **ppszTokens = NULL;
+    char *pszTmp = NULL;
+    const char *pszEvr = NULL;
+    Map *pMapMinVersions = NULL;
+    Solvable *pSolv;
+    Dataiterator di;
+    int i;
+
+    if(!pTdnf || !pQueueJobs || !pPool)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    ppszPackages = pTdnf->pConf->ppszMinVersions;
+    if (!ppszPackages)
+    {
+        goto cleanup;
+    }
+
+    dwError = TDNFAllocateMemory(
+                          1,
+                          sizeof(Map),
+                          (void**)&pMapMinVersions);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    map_init(pMapMinVersions, pPool->nsolvables);
+
+    for (i = 0; ppszPackages && ppszPackages[i]; i++)
+    {
+        pszPkg = ppszPackages[i];
```

Comment:
same here, `char *pszPkg` and please check for other variables whose cope can be reduced. Seems like too many local variables at top.